### ISSUE TITLE
ActiveProfiles doesn't support ${some.env:default} Property Expressions anymore?

### ISSUE THREAD

**Issue by icyerasor** [spring-projects/spring-framework#26611]
**Affects:** spring-test 5.3.3

Back with _5.2.1.RELEASE_ I could annotate a testclass with:
```
@ActiveProfiles("${testprofiles.active:test}")
```
when supplying `-Dtestprofiles.active=integrationtest` the context would load the resolved integrationtest Profile:

> org.springframework.core.env.StandardEnvironment - Activating profiles [integrationtest]
> com.iteratec.myTest: The following profiles are active: integrationtest

After upgrading to _5.3.3_ this doesn't seem to work anymore:

> org.springframework.core.env.StandardEnvironment - Activating profiles [${testprofiles.active:test}]
> com.iteratec.myTest: The following profiles are active: ${testprofiles.active:test}

and thus a fallback to no matching profile at all / base-properties happens.

Is this expected behavior or a bug?

<br />

I worked around this by using a custom ActiveProfilesResolver:
```
@ActiveProfiles(value = "test", resolver = SystemPropertyActiveProfilesResolver.class)
```

```
public class SystemPropertyActiveProfilesResolver implements ActiveProfilesResolver {

	private final DefaultActiveProfilesResolver defaultActiveProfilesResolver = new DefaultActiveProfilesResolver();


	@Override
	public String[] resolve(Class<?> testClass) {
		Assert.notNull(testClass, "Class must not be null");

		if (System.getProperties().containsKey("spring.profiles.active")) {
			final String profiles = System.getProperty("testprofiles.active");
			return profiles.split("\\s*,\\s*");
		} else {
			return defaultActiveProfilesResolver.resolve(testClass);
		}
	}
}
```

**Comment by sbrannen** [comment]
I was unable to reproduce this with Spring Framework 5.2.1.RELEASE.

In addition, I don't believe that Spring Framework ever supported property placeholders for `@ActiveProfiles`.

Though, Spring Boot Test did at one point set the `spring.profiles.active` system property which would result in nested placeholders being resolved against the Spring `Environment`.

Thus, perhaps this change in behavior is related to Spring Boot and not Spring Framework.

@wilkinsona, is this perhaps related to https://github.com/spring-projects/spring-boot/issues/19788?

**Comment by wilkinsona** [comment]
I don't think so. That issue was fixed in Spring Boot 2.1 which uses Framework 5.1. @icyerasor If you're using Spring Boot, what versions were involved?

**Comment by sbrannen** [comment]
@wilkinsona, on a hunch, I'd say https://github.com/spring-projects/spring-boot/commit/49921d65acd29a9704c2dd54ef405d6654dcb321 introduced the change in behavior as an unintentional side effect.

I'll see if I can throw together some tests to prove or disprove it.

**Comment by sbrannen** [comment]
Using Spring Boot `2.3.9.RELEASE` as-is, property placeholders in profiles configured via `@ActiveProfiles` are replaced.

The same is true for Spring Boot `2.3.9.RELEASE` using Spring Framework `5.3.4`.

Property placeholder replacement stops working with Spring Boot `2.4.0` (and continues not to work with Boot `2.4.3`).

Thus something must have changed in Spring Boot 2.4.

@wilkinsona and @philwebb, would you mind taking this over in Boot's issue tracker?

**Comment by wilkinsona** [comment]
If @icyerasor confirms they're using Boot then we should definitely transfer this over. If they're not then I guess we have two changes to track down somehow and we should probably open a separate Boot issue.

**Comment by icyerasor** [comment]
Hi there, 
sorry was on a leave over the weekend..
@wilkinsona yes we're using Boot / spring-boot-starter-test (2.4.2).

@sbrannen sorry for the confusion; should have probably stated this to begin with, but just saw @ActiveProfile from -test and thus blamed it 😅
